src/traces/scatterregl/plot.js

```diff
@@ -0,0 +1,575 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+var Axes = require('../../plots/cartesian/axes');
+var ErrorBars = require('../../components/errorbars');
+var str2RGBArray = require('../../lib/str2rgbarray');
+var formatColor = require('../../lib/gl_format_color');
+var subTypes = require('../scatter/subtypes');
+var makeBubbleSizeFn = require('../scatter/make_bubble_size_func');
+var getTraceColor = require('../scatter/get_trace_color');
+var DASHES = require('../../constants/gl2d_dashes');
+var createScatter = require('regl-scatter2d');
+var createLine = require('regl-line2d');
+var Drawing = require('../../components/drawing');
+var svgSdf = require('svg-path-sdf');
+var createRegl = require('regl');
+
+var DESELECTDIM = 0.2;
+var TRANSPARENT = [0, 0, 0, 0];
+
+// tables with symbol SDF values
+var SYMBOL_SDF_SIZE = 200;
+var SYMBOL_SIZE = 20;
+var SYMBOL_STROKE = SYMBOL_SIZE / 20;
+var SYMBOL_SDF = {};
+var SYMBOL_SVG_CIRCLE = Drawing.symbolFuncs[0](SYMBOL_SIZE * 0.05);
+
+var convertNumber, convertColorBase;
+
+
+module.exports = createLineWithMarkers;
+
+
+function createLineWithMarkers(container, plotinfo, cdscatter) {
+    var layout = container._fullLayout;
+
+    var subplotObj = layout._plots.xy;
+    var scatter = subplotObj._scattergl;
+
+    // create regl-scatter, if not defined
+    if(scatter === undefined) {
+        // TODO: enhance picking
+        // TODO: figure out if there is a way to detect only new passed options
+
+        scatter = subplotObj._scattergl = new ScatterScene(container);
+    }
+
+    container._fullData.forEach(function(data, i) {
+        scatter.update(cdscatter[i]);
+    });
+
+    return scatter;
+}
+
+function ScatterScene(container) {
+    if(!(this instanceof ScatterScene)) return new ScatterScene(container);
```

Comment:
No need to be convenient here. It's always going to be an internal API, so let's just use _new ScatterScene_